cube/src/main/java/zone/cogni/asquare/cube/shacl/ShaclToConversionProfile.java

```diff
@@ -131,20 +131,39 @@ private CompactConversionProfile.Attribute calculateAttribute(@Nonnull CompactCo
     attribute.setProperty(calculateProperty(propertyShape));
     attribute.setSingle(calculateSingle(propertyShape));
     attribute.setType(calculateAttributeType(propertyShape));
-    attribute.setInverse(false);
+    attribute.setInverse(isInversePropertyShape(propertyShape));
     return attribute;
   }
 
+  private boolean isInversePropertyShape(Resource propertyShape) {
+    boolean hasPath = propertyShape.hasProperty(Shacl.path);
+    boolean hasInversePath = propertyShape.hasProperty(Shacl.inversePath);
+
+    if (hasPath && hasInversePath)
+      throw new RuntimeException("both 'path' or 'inversePath' on property '" + propertyShape.getURI() + "'");
+
+    if (hasPath) return false;
+    if (hasInversePath) return true;
+
+    throw new RuntimeException("missing 'path' or 'inversePath' on property '" + propertyShape.getURI() + "'");
+  }
+
   @Nonnull
   private String calculatePathId(@Nonnull CompactConversionProfile profile,
                                  @Nonnull Resource propertyShape) {
-    Resource resource = propertyShape.getPropertyResourceValue(Shacl.path);
+    Resource resource = getPathOrInversePathResource(propertyShape);
     return convertResourceToId(profile, resource);
   }
 
   @Nonnull
   private String calculateProperty(@Nonnull Resource propertyShape) {
-    return propertyShape.getPropertyResourceValue(Shacl.path).getURI();
+    return getPathOrInversePathResource(propertyShape).getURI();
```

Comment:
are we sure the returned value is non-null?

Reply:
yes. because in other case it will return an exception now.